### PR TITLE
Added jdbc-relational-duality-example

### PR DESCRIPTION
**Pull request by aberinnj**
Please **review** and merge my code samples for my article on JDBC, JSON, Relational Duality views and DB23ai

@gvenzl @Kuassim

### CODE REVIEW

**Comment by aberinnj** [comment]
@Kuassim Please review and help merge

**Comment by aberinnj** [comment]
@Kuassim oh found something to fix! `pom.xml` file reflects older JDBC version

File: java/jdbc-relational-duality-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <maven.compiler.source>22</maven.compiler.source>
+        <maven.compiler.target>22</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.oracle.database.jdbc</groupId>
+            <artifactId>ojdbc11-production</artifactId>
+            <version>23.2.0.0</version>
```
Comment (jeandelavarene) [inline java/jdbc-relational-duality-example/pom.xml:21]: Please use 23..4
https://mvnrepository.com/artifact/com.oracle.database.jdbc/ojdbc11-production/23.4.0.24.05

File: java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+package org.oracle;
+
+import oracle.sql.json.OracleJsonArray;
+import oracle.sql.json.OracleJsonFactory;
+import oracle.sql.json.OracleJsonObject;
+
+
+public class Main {
+		private static DatabaseConfig pds;
```
Comment (jeandelavarene) [inline java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java:28]: Please use consistent indentation across java files

**Comment by aberinnj** [comment]
@jeandelavarene @Kuassim please re-review

File: java/jdbc-relational-duality-example/src/main/java/org/oracle/DatabaseConfig.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+import oracle.jdbc.OracleConnection;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+
+import java.sql.SQLException;
+public class DatabaseConfig {
+
+
+    private static final String DB_USER = System.getenv("db.user");
```
Comment (jeandelavarene) [inline java/jdbc-relational-duality-example/src/main/java/org/oracle/DatabaseConfig.java:33]: Please use the same indentation in all your Java files to make the reading experience easier. For example, you can use 2 space characters.

File: java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+package org.oracle;
+
+import oracle.sql.json.OracleJsonArray;
+import oracle.sql.json.OracleJsonFactory;
+import oracle.sql.json.OracleJsonObject;
+
+
+public class Main {
+	private static DatabaseConfig pds;
```
Comment (jeandelavarene) [inline java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java:28]: Please use the same indentation in all your Java files to make the reading experience easier. For example, you can use 2 space characters.

File: java/jdbc-relational-duality-example/src/main/java/org/oracle/DatabaseConfig.java (hunk 3 above)
Comment (jeandelavarene) [reply java/jdbc-relational-duality-example/src/main/java/org/oracle/DatabaseConfig.java:33]: Also avoid using tabs.

File: java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java (hunk 4 above)
Comment (aberinnj) [reply java/jdbc-relational-duality-example/src/main/java/org/oracle/Main.java]: OK I have fixed it to 2 spaces

**Comment by aberinnj** [comment]
@jeandelavarene @Kuassim Please re-review, I used spaces instead for indentation

**Review by jeandelavarene — APPROVED** [review]

**Pull request merged**